### ISSUE TITLE
GUI tool for file renaming

### ISSUE THREAD

**Issue by oakmac** [oakmac/open-ideas#7]
There are many powerful command line tools for finding and renaming files ([ack](http://beyondgrep.com), [sed](https://en.wikipedia.org/wiki/Sed), [grep](https://en.wikipedia.org/wiki/Grep), etc). Additionally, these tools can be combined using [pipes](https://en.wikipedia.org/wiki/Pipeline_%28Unix%29) for even more power and flexibility.

Most computer users are unfamiliar or uncomfortable on the command-line. Even users comfortable on the command line do not necessarily have the arguments for these commands memorized: they are complex and esoteric.

It would be nice to have a powerful cross-platform GUI tool for file renaming. I think you could achieve ~95% of the power of command line tools with a UI design that required little to no training in order to use.

**Comment by shaunlebron** [comment]
> It would be nice to have a powerful cross-platform GUI tool for file renaming

you mean batch file renaming using some type of pattern matching?

Rename `*.foo` -> `*.bar` ?

**Comment by oakmac** [comment]
Yes, batch file searching and renaming using pattern matching or regular expressions. Automatic preview of the change before it's made, etc.
